iceoryx_hoofs/test/moduletests/test_container_vector.cpp

```diff
@@ -960,6 +963,86 @@ TEST_F(vector_test, EraseOfMiddleElementCallsDTorAndMove)
     EXPECT_THAT(moveAssignment, Eq(2U));
 }
 
+TEST_F(vector_test, AccessOfNonExistingElementOnEmptyVectorLeadTermination)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "31a4f0fb-31dd-4119-baba-31efab42c42b");
+    constexpr uint64_t CAPACITY{31U};
+    constexpr uint64_t DEFAULT_VALUE{69U};
+    constexpr uint64_t ACCESS_OFFSET{CAPACITY + 1U};
```

Comment:
We should use some index `1 <= ACCESS_OFFSET < CAPACITY` here as the offset. I.e. access a position that is within capacity limits. Say `1`.

Generally I would recommend using a value that depends on size (after vector creation of course), like ```const uint64_t accessOffset{sut.size() + 1};```.

This holds for all tests. We are more interested in overruns of size (say by 1) than overruns of the whole capacity. 

Of course, we may miss any error with a finite small test set, but the art is to pick the most useful values to check. I would argue `size + 1` is one of them, and more so than `capacity + 1`.

@ibrhmkuru We should consider this for all tests of this sort.

Reply:
@MatthiasKillat You're definitely right. It is better to test for `size+1`. I've updated the tests accordingly.